pybandits/benchmark_models.py

```
@@ -0,0 +1,577 @@
+            rng = np.random.default_rng(seed)
+            model = spec.factory(n_actions=n_actions, random_seed=seed, **model_kwargs)
+            action_ids = _action_ids(n_actions)
+            actions = [action_ids[i % n_actions] for i in range(n_samples)]
+            rewards = make_rewards(spec, n_samples, rng)
+            context = rng.normal(size=(n_samples, n_features)) if spec.contextual else None
+
+            chunk = batch_size or n_samples
+            slices = [slice(i, min(i + chunk, n_samples)) for i in range(0, n_samples, chunk)]
+
+            def call() -> None:
+                for sl in slices:
```

Ruff (0.15.21)

[warning] Function definition does not bind loop variable `slices`

(B023)